### PR TITLE
Start explanation of WABuR.

### PR DESCRIPTION
**Pull request by ohler55**
Explain what application WABuR is best suited for. Also expand the benchmark data to include Rails and Sinatra. Both insert and fetch should be included in the benchmarks.

### CODE REVIEW

**Comment by ohler55** [comment]
Ok, ready for review and suggestions.

File: README.md [hunk 1]
```diff
@@ -5,27 +5,117 @@
 [![Gem Version](https://badge.fury.io/rb/wabur.svg)](https://rubygems.org/gems/wabur)
 [![Gem](https://img.shields.io/gem/dt/wabur.svg)](https://rubygems.org/gems/wabur)
 
-WABuR is a Web Application Builder using Ruby. It employs a modern NoSQL JSON
-data store and a single-page UI using JavaScript. The best part is that it is
-simple and very fast, hitting over 200,000 fetches a second with a Ruby core!
+WABuR is a Web Application Builder using Ruby. It is easy to use, taking just
+a minute to create a hello world web application and it is _FAST_, hitting
```
Comment (ashmaroli) [inline README.md]: ```..create a `Hello World` web application```

File: README.md [hunk 2]
```diff
@@ -5,27 +5,117 @@
...
+data store and a single-page UI using JavaScript.
 
 It is pluggable and extendable in many ways to allow new additions,
 alternative databases, and any number of UIs.
 
-A natural question is *"What about Rails?"*. Rails is well established and has
-a huge user base. WABuR is not a replacement for Rails. It is an alternative
-for those who want to explore using JSON databases with a single-page dynamic
-JavaScript UI.
+## Simple to use Quick Start
```
Comment (ashmaroli) [inline README.md]: Isn't simply `Quick Start` just as effective?

File: README.md [hunk 3]
```diff
@@ -5,27 +5,117 @@
...
+With WABuR you are up and running in minutes with only one file to modify to
+specify attributes. It doesn't get any simplier. Heres how.
+
+Install the wabur gem.
+
+```
+$ gem install wabur
+```
+
+Create a new project and cd into the directory.
```
Comment (ashmaroli) [inline README.md]: ```..and `cd` into..```

File: README.md [hunk 4]
```diff
@@ -5,27 +5,117 @@
...
+$ cd blog
+```
 
-## Try It!
+Define attributes for the data elements along with the attributes that will be
+displayed in a list view. Open `lib/ui_controller.rb` and modify by adding two
+lines and changing one.
+
+```ruby
+# encoding: UTF-8
```
Comment (ashmaroli) [inline README.md]: `# encoding: UTF-8` can be left out in actually.. The interpreter encodes to UTF-8 by default since Ruby 2.0

File: README.md [hunk 5]
```diff
@@ -36,14 +126,16 @@ and lets make something awesome together.
...
 These are the simple guidelines for contributing.
 
-1. Coordinate with me first before getting started to avoid duplication of
+1. Take a look at the [architecture page](pages/Architecture.md) and the source code.
+
+2. Coordinate with me first before getting started to avoid duplication of
    effort or implementing something in conflict with the plans.
 
-2. Branch off the develop branch and submit a PR.
+3. Branch off the develop branch and submit a PR.
```
Comment (ashmaroli) [inline README.md]: ```..off the `develop` branch..```

File: benchmarks/README.md [hunk 6]
```diff
@@ -1,115 +1,136 @@
...
 
-To run the benchmarks start **`opod`** with the `opo/bench.conf` file. This
-turns off the verbosity on `opod` and on the sample Ruby app.
+| Runner        | Read Throughput | Read Latency | Create Throughput | Create Latency | Setup time |
+| ------------- | --------------- | ------------ | ----------------- | -------------- | ---------- |
+| WAB Pure Ruby | 2.8K Reads/sec  | 1.4 msecs    | 2.2K Creates/sec  | 1.8 msecs      |  1 minute  |
+| OpO-Rub       | 212K Reads/sec  | 0.09 msecs   | 134K Creates/sec  | 0.15 msecs     |  1 minute  |
+| OpO Direct    | 364K Reads/sec  | 0.05 msecs   | 157K Creates/sec  | 0.13 msecs     |  1 minute  |
+| Ruby on Rails |  123 Reads/sec  | 175 msecs    | ---- Creates/sec  | ---- msecs     | 20 minutes |
+| Sinatra       | 1.5K Reads/sec  |  13 msecs    | ---- Creates/sec  | ---- msecs     | 60 minutes |
```
Comment (ashmaroli) [inline benchmarks/README.md]: Sinatra took an hour to set up?

File: benchmarks/README.md [hunk 7]
```diff
@@ -1,115 +1,136 @@
...
-  * By going directly to the database with the path
-    `http://localhost:6363/json/000000000000000b` which uses the ref in hex to
-    identify the record. This is interesting in that it shows the overhead of
-    the calls to the Ruby Controller.
+## Details
 
-Now that a record has been created, the benchmarks can be run.
+The benchmarks compare two different WABuR runners along with a Ruby on Rails
+sample and a Sinatra sample. An OpO direct comparison is also given for
+comparisone although it is a direct database access without Ruby code being
```
Comment (ashmaroli) [inline benchmarks/README.md]: typo: :point_right: `comparisone`. Better please reword the entire sentence..

File: benchmarks/README.md [hunk 8]
```diff
@@ -1,115 +1,136 @@
...
+invoked. For all the runs at least the read operations were run. Setup times
```
Comment (ashmaroli) [inline benchmarks/README.md]: ```For all the runs, at least..```

File: pages/Architecture.md [hunk 9]
```diff
@@ -158,12 +158,21 @@ API. From the view perspective a REST over HTTP is used.
...
 #### Just Data
 
-  - just data and data helpers (get, set, inspect, to_s)
-   - nothing like to_json
-   - reasoning it that different uses require different behavior
-    - different stores, view, processing, etc
-   - use of delgate is encouraged
-
+WABuR keeps data and behavior separate. The data portion of a system is
+represented by _WAB::Data_ class. Behavior is implemented by support
```
Comment (ashmaroli) [inline pages/Architecture.md]: use back-ticks instead of italicizing Ruby (in this entire section.)

File: README.md (hunk 2 above)
Comment (ohler55) [reply README.md]: probably so.

File: benchmarks/README.md (hunk 6 above)
Comment (ohler55) [reply benchmarks/README.md]: Sinatra gives you nothing other than routes and ERB. It has no storage. You have to set up your own separate pages for view, create, edit, etc. If you want a simple REST like API Sinatra does not provide any of that.

**Comment by ohler55** [comment]
Thanks. You are very good at catching the details.

**Comment by ashmaroli** [comment]
> You are very good at catching the details.

Hehe.. its probably the resolution on my screen.. :wink: ..and the fact that a fresh set of :eyes: helps in seeing errors.. Hence the advantage of a peer-review :smiley:

**Comment by ohler55** [comment]
I'm sure both help. :-)

**Comment by ohler55** [comment]
What do you think of focussing on the quick start and performance on the landing page? I figured a to the point first impression would be best.

File: README.md [hunk 10]
```diff
@@ -5,27 +5,115 @@
...
 alternative databases, and any number of UIs.
 
-A natural question is *"What about Rails?"*. Rails is well established and has
-a huge user base. WABuR is not a replacement for Rails. It is an alternative
-for those who want to explore using JSON databases with a single-page dynamic
-JavaScript UI.
+## Quick Start
+
+With WABuR you are up and running in minutes with only one file to modify to
+specify attributes. It doesn't get any simplier. Heres how.
```
Comment (ashmaroli) [inline README.md]: missed this one earlier :point_right: `simplier` :smile:

**Comment by ashmaroli** [comment]
> focussing on the quick start and performance on the landing page?

that's a nice idea..

**Pull request merged**